### PR TITLE
Fix conf dir link failing

### LINKED ISSUE #320: Can't configure DSE's config directory
When using the recipe to install DSE, the value of `node['cassandra']['conf_dir']` has to be customized to `/etc/dse/cassandra` to match the installation location of DSE. However, that causes the following error:

```
ERROR: link[/etc/dse/cassandra] (cassandra-dse::datastax line 174) had an error: Errno::EISDIR: Is a directory - /etc/dse/cassandra
```

That's due to the recipe trying to symlink the configured conf_dir to the default conf_dir and failing because the configured conf_dir already exists (the DSE installation created it).

Either this link should not be done, or the symlink should be the other way around.

### PR DESCRIPTION
**Pull request by rehevkor5**
Fixes #320.

### CODE REVIEW

File: recipes/datastax.rb [hunk 1]
```diff
@@ -191,7 +191,10 @@
     to node.default['cassandra']['conf_dir']
     owner node['cassandra']['user']
     group node['cassandra']['group']
-    not_if { node['cassandra']['conf_dir'] == node.default['cassandra']['conf_dir'] }
+    not_if {
+      node['cassandra']['conf_dir'] == node.default['cassandra']['conf_dir'] ||
+          ::File.exist?(node['cassandra']['conf_dir'])
```
Comment (rehevkor5) [inline recipes/datastax.rb]: This is the fix.

File: spec/default_spec.rb [hunk 2]
```diff
@@ -177,6 +177,82 @@
...
+      expect(chef_run).to create_directory('/etc/mycassandra/conf').with(
+          owner: 'cassandra',
+          group: 'cassandra',
+          recursive: true,
+          mode: '0755'
+      )
+    end
+
+    it 'does not create a symlink from conf_dir to /etc/mycassandra/conf' do
+      puts 'CHECKING'
```
Comment (michaelklishin) [inline spec/default_spec.rb]: Please remove these.

**Comment by michaelklishin** [comment]
Thank you, looks reasonable and has tests 👍. Let's remove the traces and then I'll take one last look.

File: spec/default_spec.rb (hunk 2 above)
Comment (rehevkor5) [inline spec/default_spec.rb]: My mistake! I pushed an amended commit.

**Comment by michaelklishin** [comment]
Thank you!

**Pull request merged**